io.github.lpub3d.LPub3D.json

```diff
@@ -0,0 +1,183 @@
+{
+  "id": "io.github.lpub3d.LPub3D",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.15",
+  "sdk": "org.kde.Sdk",
+  "command": "lpub3d24",
+  "rename-icon": "lpub3d",
+  "rename-desktop-file": "lpub3d.desktop",
+  "rename-appdata-file": "io.github.lpub3d.appdata.xml",
+  "build-options": {
+    "env": {
+      "LP3D_3RD_PARTY_DIR": "/stage/lpub3d_linux_3rdparty",
+      "LP3D_DIST_DIR_PATH": "/stage/lpub3d_linux_3rdparty",
+      "LDRAWDIR": "/stage/lpub3d_linux_3rdparty/ldraw",
+      "PLATFORM_CODE": "osl",
+      "TARGET_CPU": "$(uname -m)",
+      "PLATFORM_VER": "$(. /etc/os-release 2>/dev/null; [ -n \"$VERSION_ID\" ] && echo $VERSION_ID || true)"
+    }
+  },
+  "cleanup": [
+    "/stage"
+  ],
+  "finish-args": [
+    "--device=dri",
+    "--share=ipc",
+    "--socket=x11",
+    "--filesystem=home",
+    "--env=TMPDIR=/var/tmp",
+    "--env=QT_ENABLE_HIGHDPI_SCALING=1"
+  ],
+  "modules": [
+    {
+      "name": "part-archives",
+      "buildsystem": "simple",
+      "build-commands": [
+        "unzip -d ${LP3D_3RD_PARTY_DIR} -q /stage/lpub3d_linux_3rdparty/complete.zip",
+        "if test -d ${LP3D_3RD_PARTY_DIR}/ldraw; then echo 'LDraw library extracted.'; fi"
+      ],
+      "sources": [
+        {
+          "type": "file",
+          "dest-filename": "complete.zip",
+          "url": "https://github.com/trevorsandy/lpub3d_libs/releases/download/v1.0.1/complete-20-03.zip",
+          "sha256": "85cf747b5923f18a8d0f37d0cef647636d70ddfeda5de49ea019e7e440f9e4e0"
+        },
+        {
+          "type": "file",
+          "dest-filename": "lpub3dldrawunf.zip",
+          "url": "https://github.com/trevorsandy/lpub3d_libs/releases/download/v1.0.1/lpub3dldrawunf-01-04-21.zip",
+          "sha256": "f0266006c4b7114e4f4719801696fbe8780cf9d767e948fb6478c752989cbee8"
+        },
+        {
+          "type": "file",
+          "url": "https://github.com/trevorsandy/lpub3d_libs/releases/download/v1.0.1/tenteparts.zip",
+          "sha256": "7e1a68490b1be11bddcfb3180151b42ffbb85ce3aa8ec3f7431bee5c04cc1ac5"
+        },
+        {
+          "type": "file",
+          "url": "https://github.com/trevorsandy/lpub3d_libs/releases/download/v1.0.1/vexiqparts.zip",
+          "sha256": "e2f4114c8ff6bfe7e5a43b6c88fb47f32328ddb147578456ec0825aa76803b20"
+        }
+      ],
+      "post-install": [
+        "mv -f complete-20-03.zip ${LP3D_3RD_PARTY_DIR}/complete.zip",      
+        "mv -f lpub3dldrawunf-01-04-21.zip ${LP3D_3RD_PARTY_DIR}/lpub3dldrawunf.zip",
```

Comment:
even if `$LP3D_3RD_PARTY_DIR` is valid, you rename the files with `dest-filename` (good) so you should use the renamed filename here.

Reply:
> even if $LP3D_3RD_PARTY_DIR is valid, you rename the files with dest-filename (good)

Oops, forgot about that one.

> I'm not sure I understand what this staging is for

The staging folder is internal to how LPub3D captures the renderer and parts library artefacts. What I need from the flatpack environment is a writable location - that's it, but I'm struggling to determine where I can put my staging folder and be able to pass a full path to the renderers and LPub3D?

Do you see any restriction with using `"LP3D_3RD_PARTY_DIR": "$(realpath ./lpub3d_linux_3rdparty)"` ?

Cheers,

Reply:
it won't work. I'm not even sure shell substitution is even done there anyway.

`${FLATPAK_DEST}/lpub3d_linux_3rdparty` should work, but you'll need to create it early and then add `/lpub3d_linux_3rdparty` in the `cleanup` section.